src/app/recipes/store/recipe.effect.ts

```diff
@@ -1,41 +1,55 @@
 import { Actions, Effect, ofType } from '@ngrx/effects';
-import { switchMap, map, withLatestFrom, tap } from 'rxjs/operators';
+import { switchMap, map, withLatestFrom, tap, catchError } from 'rxjs/operators';
 import { Injectable } from '@angular/core';
 import { HttpClient, HttpParams } from '@angular/common/http';
 import { environment } from 'src/environments/environment';
+import { Store, select } from '@ngrx/store';
+import { of } from 'rxjs';
 
 import * as RecipesActions from './recipe.actions';
 import { Recipe } from '../recipe.model';
 import * as fromApp from '../../store/app.reducer';
-import { Store, select } from '@ngrx/store';
+import * as RecipesSelectors from './recipe.selectors'
+
+const handleError = (errRes: any) => {
+  console.log(errRes);
+  
+  return of(new RecipesActions.CrudFail(errRes.error.error));
+};
 
 @Injectable()
 export class RecipeEffects {
   @Effect()
   fetchRecipes = this.actions$.pipe(
     ofType(RecipesActions.FETCH_RECIPES),
-    switchMap(() => {
+    switchMap((actionData: RecipesActions.FetchRecipes) => {
       const url = window.location.href;
       let httpParams = new HttpParams()
-        .set('startItem', '0')
-        .append('limit', '5');
+        .set('startItem', actionData.payload.startItem.toString())
+        .append('limit', actionData.payload.limit.toString());
       if (url.includes('?')) {
         httpParams = new HttpParams({ fromString: url.split('?')[1] });
       }
-      return this.http.get<Recipe[]>(environment.DataBaseUrl + 'recipes', {
-        params: httpParams,
-      });
+      return this.http.get<{ recipes: Recipe[]; maxRecipes: number }>(
+        environment.DataBaseUrl + 'recipes',
+        {
+          params: httpParams,
+        }
+      );
     }),
-    map((recipes) => {
-      return recipes.map((recipe) => {
-        return {
-          ...recipe,
-          ingredients: recipe.ingredients ? recipe.ingredients : [],
-        };
-      });
+    map((postData) => {
+      return {
+        recipes: postData.recipes.map((recipe) => {
+          return {
+            ...recipe,
+            ingredients: recipe.ingredients ? recipe.ingredients : [],
+          };
+        }),
+        maxRecipes: postData.maxRecipes,
+      };
```

Comment:
As I see there no transformations. For what you implement this `map`?

Reply:
Fixed. Yes, now it is not needed. I just used firebase like server at the beginning, there were problems with this ingredients.